apps/system/js/screen_manager.js

```diff
@@ -206,6 +206,14 @@
 
         case 'sleep':
           this.turnScreenOff(true, 'powerkey');
+          var lock_enabled = window.lockScreen.enabled;
+          var passcode_enabled = window.lockScreen.passCodeEnabled;
```

Comment:
And it's improper to read another component's status through no the `request` method of System/Service. This would create a tight coupling that hinders us to modify components individually in the future.